### PR TITLE
Adding hackbench test to avocado[V2]

### PR DESCRIPTION
**Pull request by PraveenPenguin**
Signed-off-by: praveen@linux.vnet.ibm.com
added hackbench Test case
v1 #76 

```
V2: removed extra  variables 
V2: set threshold time in yaml file 
```

### CODE REVIEW

File: perf/hackbench.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        if 'CC' in os.environ:
+            cc = '$CC'
+        else:
+            cc = 'cc'
+        process.system('%s  hackbench.c -o hackbench -lpthread' % cc)
+
+    def test(self):
+
+        time_value = self.params.get('time_val', default=None)
+        num_groups = self.params.get('num_groups', default=10)
```
Comment (apahim) [inline perf/hackbench.py:65]: If you do the `self.params.get()` on `setUp()`, the parameters will be available for all tests. Ok, you have only one here, but still, it's a good practice.

File: perf/hackbench.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        hackbench_bin = os.path.join(self.srcdir, 'hackbench')
+        cmd = '%s %s' % (hackbench_bin, num_groups)
+        self.results = process.system_output(cmd, shell=True)
+        perf_json = {}
+        for line in self.results.split('\n'):
+            if line.startswith('Time:'):
+                time_val = line.split()[1]
+                time = time_val
```
Comment (apahim) [inline perf/hackbench.py:73]: What about `time = line.split()[1]`?
Also, this `time`/`time_val`/`time_value` is very confusing. I'd call:
`time_limit = self.params.get('time_limit', default=None)`
and
`time_spent = line.split()[1]`
And use only those two terms.

File: perf/hackbench.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                perf_json = {'time': time_val}
+        output_path = os.path.join(self.outputdir, "perf.json")
+        json.dump(perf_json, open(output_path, "w"))
+        self.log.info("Time Taken:" + time)
+        if time_value:
+            if time_value != time:
```
Comment (apahim) [inline perf/hackbench.py:79]: I don't get this. Shouldn't the check be `if time >= time_value:` ?

File: perf/hackbench.py (hunk 2 above)
Comment (PraveenPenguin) [inline perf/hackbench.py:73]: thanks for looking in to this , sure that make sense will do

File: perf/hackbench.py (hunk 3 above)
Comment (PraveenPenguin) [inline perf/hackbench.py:79]: @apahim sure that make sense

Thanks

File: perf/hackbench.py (hunk 1 above)
Comment (PraveenPenguin) [inline perf/hackbench.py:65]: @apahim 
thanks for pointing on this , sure that make sense

**Comment by PraveenPenguin** [comment]
@apahim added new PR #117  with change please have a look

Thanks

**Pull request closed without merging**